core/store/src/trie/mem/split.rs

```diff
@@ -325,4 +334,326 @@ mod tests {
         let threshold = 1050;
         assert_eq!(find_middle_child(&children_mem_usage, threshold), Some((10, 1000)));
     }
+
+    #[test]
+    fn nibble_ops() {
+        assert_eq!(byte_to_nibbles(0x00), (0x00, 0x00));
+        assert_eq!(byte_to_nibbles(0x01), (0x00, 0x01));
+        assert_eq!(byte_to_nibbles(0x10), (0x01, 0x00));
+        assert_eq!(byte_to_nibbles(0x11), (0x01, 0x01));
+
+        assert_eq!(nibbles_to_bytes(&[]), &[] as &[u8]);
+        assert_eq!(nibbles_to_bytes(&[0x01]), &[] as &[u8]);
+        assert_eq!(nibbles_to_bytes(&[0x00, 0x01]), &[0x01]);
+        assert_eq!(nibbles_to_bytes(&[0x01, 0x02, 0x03]), &[0x12]);
+    }
+
+    fn new_leaf(arena: &mut STArena, value: &[u8], extension_nibbles: &[u8]) -> MemTrieNodeId {
```

Comment:
nit, here and below - it is more intuitive to pass the key first and then the value.